WordPress/gutenberg
# Writing Flow: Fixing Show + Arrow multi selection across Spacer

## What?

Fix issue happens on https://github.com/WordPress/gutenberg/issues/53674

## Why?

- Shift + down-arrow key to select across spacer block are selecting too many blocks
![260688744-ebacc902-7343-4824-a748-e83474302d91](https://github.com/WordPress/gutenberg/assets/10071857/c7674dad-0f26-4b1a-95b7-018f725737f9)


## How?

- Spacer is skipped by browser because of its empty content and its children are hidden from the DOM
   - More on [multi-selection based on the native selection across blocks](https://github.com/WordPress/gutenberg/blob/77453a104d5c438ecf2aa21bc7bceef59b0f0d78/packages/block-editor/src/components/writing-flow/use-selection-observer.js#L87)
- 1st attempt is to update Spacer handle visibility instead of setting `display: none` https://github.com/WordPress/gutenberg/issues/53674#issuecomment-1689594974
- 2nd attempt insert BR inside Spacer block

## Testing Instructions




- Run `npm run dev`
- Edit a post
- Add a Text Block - Spacer - Text Block
- Select first text block then press Shift + Arrow down
- Expect to select the first Text Block + Spacer as 2 selected items instead of all 3 blocks

### Testing Instructions for Keyboard


## Screenshots or screencast 
- Check 1st attempt recordings on https://github.com/WordPress/gutenberg/issues/53674#issuecomment-1689594974
- BR approach fix https://github.com/WordPress/gutenberg/issues/53674#issuecomment-1694643333

cc: @jasmussen
